llvm/lib/Transforms/Instrumentation/NumericalStabilitySanitizer.cpp

```diff
@@ -493,6 +494,103 @@ class ValueToShadowMap {
   DenseMap<Value *, Value *> Map;
 };
 
+// Base class for handling some details of __nsan_* functions
+class NsanInstrumentationFunction {
```

Comment:
The classes share the interface, but they are not used polymorphically, so the base class doesn't do much  / I'd simplify things by removing it.

Reply:
Done in the latest commit